### PR TITLE
[feature/#424] Implement Sponsor List and Design on SponsorsScreen

### LINKED ISSUE #424: Implement Sponsor List and Design on SponsorsScreen
**Idea Description**
Create and display the sponsor list on the SponsorsScreen, and apply the corresponding design as shown on [Figma](https://www.figma.com/file/MbElhCEnjqnuodmvwabh9K/DroidKaigi-2023-App-UI). The task includes:
   - Implementing the sponsor list.
   - Placing the list's Composable functions in the `section` package. More information is available [here](https://github.com/DroidKaigi/conference-app-2023#section).
   - Creating the UiState as needed.
   - Applying the design to match the provided Figma layout.

**Reference images and links**
image

### PR DESCRIPTION
**Pull request by l2hyunwoo**
## Issue
- close #424 

## Overview (Required)
- Add SponsorList Composable in section package
- Add SponsorItem, SponsorHeader in component package
- Separate sponsors by plan at SponsorsUiState

### CODE REVIEW

**Comment by takahirom** [comment]
Maybe we should adjust the Fake's id  or something? 👀 
>java.lang.IllegalArgumentException: Key "DroidKaigi" was already used. If you are using LazyColumn/Row please make sure you provide a unique key for each item.

**Comment by l2hyunwoo** [comment]
@takahirom It caused by duplicated id from fake sponsor list's item. I think build/test will be fixed.

**Comment by takahirom** [comment]
We have to change "#424" to "%23424" in url

Snapshot diff report
| File name | Image |
|-------|-------|
| [SponsorsScreenTest.c<br>heckLaunchShot_compa<br>re.png](https://github.com/DroidKaigi/conference-app-2023/blob/companion_feature/%23424/./screenshot-diff/feature/sponsors/build/outputs/roborazzi/SponsorsScreenTest.checkLaunchShot_compare.png) |  |
| [io.github.droidkaigi<br>.confsched2023.spons<br>ors.section_null_Def<br>aultGroup_SponsorLis<br>tPreview_0_null_comp<br>are.png](https://github.com/DroidKaigi/conference-app-2023/blob/companion_feature/%23424/./screenshot-diff/app-android/build/outputs/roborazzi/io.github.droidkaigi.confsched2023.sponsors.section_null_DefaultGroup_SponsorListPreview_0_null_compare.png) |  |
| [io.github.droidkaigi<br>.confsched2023.spons<br>ors.component_null_D<br>efaultGroup_SponsorI<br>temPreview_0_null_co<br>mpare.png](https://github.com/DroidKaigi/conference-app-2023/blob/companion_feature/%23424/./screenshot-diff/app-android/build/outputs/roborazzi/io.github.droidkaigi.confsched2023.sponsors.component_null_DefaultGroup_SponsorItemPreview_0_null_compare.png) |  |
| [KaigiAppTest.checkNa<br>vigateToSponsorsShot<br>_compare.png](https://github.com/DroidKaigi/conference-app-2023/blob/companion_feature/%23424/./screenshot-diff/app-android/build/outputs/roborazzi/KaigiAppTest.checkNavigateToSponsorsShot_compare.png) |  |
| [io.github.droidkaigi<br>.confsched2023.spons<br>ors.component_null_D<br>efaultGroup_SponsorH<br>eaderPreview_0_null_<br>compare.png](https://github.com/DroidKaigi/conference-app-2023/blob/companion_feature/%23424/./screenshot-diff/app-android/build/outputs/roborazzi/io.github.droidkaigi.confsched2023.sponsors.component_null_DefaultGroup_SponsorHeaderPreview_0_null_compare.png) |  |

File: feature/sponsors/src/main/java/io/github/droidkaigi/confsched2023/sponsors/SponsorsScreen.kt [hunk 1]
```diff
@@ -51,35 +62,59 @@ fun SponsorsScreen(
...
+                    IconButton(
+                        onClick = onBackClick,
+                    ) {
+                        Icon(
+                            imageVector = Icons.Default.ArrowBack,
+                            contentDescription = "Back",
+                        )
+                    }
+                },
+                scrollBehavior = scrollBehavior,
```
Comment (takahirom) [inline feature/sponsors/src/main/java/io/github/droidkaigi/confsched2023/sponsors/SponsorsScreen.kt:103]: 👍

**Review by takahirom — APPROVED** [review]
Looks nice! Thanks for implementing this so fast!

**Pull request merged**